js/ui/editor/editor.js

```diff
@@ -178,107 +159,61 @@ const Editor = Widget.inherit({
         return false;
     },
 
-    _renderValidationState: function() {
-        const isValid = this.option('isValid') && this.option('validationStatus') !== VALIDATION_STATUS_INVALID;
-        const validationMessageMode = this.option('validationMessageMode');
-        const $element = this.$element();
+    _getValidationErrors: function() {
         let validationErrors = this.option('validationErrors');
         if(!validationErrors && this.option('validationError')) {
             validationErrors = [this.option('validationError')];
         }
-        $element.toggleClass(INVALID_CLASS, !isValid);
-        this.setAria(VALIDATION_STATUS_INVALID, !isValid || undefined);
-
-        if(!hasWindow()) {
-            return;
-        }
+        return validationErrors;
+    },
 
+    _removeValidationMessage: function() {
         if(this._$validationMessage) {
             this._$validationMessage.remove();
             this.setAria('describedby', null);
             this._$validationMessage = null;
         }
+    },
 
-        const validationErrorMessage = getValidationErrorMessage(validationErrors);
+    _toggleValidationClasses: function(isInvalid) {
+        this.$element().toggleClass(INVALID_CLASS, isInvalid);
+        this.setAria(VALIDATION_STATUS_INVALID, isInvalid || undefined);
+    },
+
+    _renderValidationState: function() {
+        const isValid = this.option('isValid') && this.option('validationStatus') !== VALIDATION_STATUS_INVALID;
+        const validationErrors = this._getValidationErrors();
+        const $element = this.$element();
 
-        if(!isValid && validationErrorMessage) {
-            this._$validationMessage = $('<div>').addClass(INVALID_MESSAGE)
-                .html(validationErrorMessage)
-                .appendTo($element);
+        this._toggleValidationClasses(!isValid);
 
-            const validationTarget = this._getValidationMessageTarget();
+        if(!hasWindow()) {
+            return;
+        }
 
-            this._validationMessage = this._createComponent(this._$validationMessage, Overlay, extend({
-                integrationOptions: {},
-                templatesRenderAsynchronously: false,
-                target: validationTarget,
-                shading: false,
-                width: 'auto',
-                height: 'auto',
+        this._removeValidationMessage();
+        if(!isValid && validationErrors) {
+            this._$validationMessage = $('<div>').appendTo($element);
+            this._validationMessage = new ValidationMessage(this._$validationMessage, extend({
```

Comment:
Confusing names

Reply:
but it's default for non-renovated DevExtreme